petastorm/reader.py

```diff
@@ -236,25 +241,22 @@ def make_batch_reader(dataset_url,
     :return: A :class:`Reader` object
     """
 
-    if dataset_url is None or not isinstance(dataset_url, six.string_types):
-        raise ValueError('dataset_url must be a string')
+    if isinstance(dataset_url_or_urls, list):
+        dataset_url_or_urls = [_check_dataset_url(url) for url in dataset_url_or_urls]
```

Comment:
Consider making `_check_dataset_url` to accept `dataset_url_or_urls` directly to avoid copy pasting this `if isinstance` code in multiple places.